Domaciukoly20/driving_cars.py

```diff
@@ -26,29 +28,28 @@ def drive(self, distance):
         self._mileage += distance
 
     def distance_left(self):
-        return self._fuel / self._economy
+        return self._fuel / self._economy * 100
 
     def fuel_up(self):
-            self._fuel += 20
+        self._fuel += 20
 
-import random
-models = ['Toyota', 'Honda', 'Ford', 'Chevrolet','Nissan']
-colors = ['blue','green','yellow','black','red','green']
+    def get_max_fuel(self):
+        return self._max_fuel
+
+
+models = ['Toyota', 'Honda', 'Ford', 'Chevrolet', 'Nissan']
+colors = ['blue', 'green', 'yellow', 'black', 'red', 'green']
 
 cars = []
 
-for i in range(10):
-    cars.append(Car(random.randrange(100), random.choice(colors), random.choice(models)))
+for _ in range(10):
+    cars.append(Car(random.randrange(4, 20), random.choice(colors), random.choice(models)))
 
 for item in cars:
     item.drive(200)
     item.fuel_up()
     item.drive(100)
 
-
-from operator import attrgetter
-
-
 max_fuel = max(cars, key=attrgetter('_fuel'))
```

Comment:
Проблема про яку я кажу ось на цьому рядку. Ти звертаешься до private attr _fuel. Так робити не слід.

Є два варіант як це зробити більш правильно.

1. Зробити get_fuel method. та використати lambda:
```suggestion
max_fuel_car = max(cars, key=lambda car: car.get_fuel())
```


2. Або варіант два. Створити `@property` fuel:

`@property`
`def fuel(self):`
      `return self._fuel`

Це фактично той же геттер. Але тоді не потрібно викликати () при використані цього методу. І тоді attrgetter буде працювати.

```suggestion
max_fuel_car = max(cars, key=attrgetter('fuel')) 
```